src/partial-evaluators/Program.js

```diff
@@ -38,8 +38,8 @@ export default function (
       partialBody.push((partialAst: any));
       if (!(potentialVal instanceof EmptyValue)) val = potentialVal;
     } else {
-      throw realm.createIntrospectionErrorThrowCompletion(
-        "function declarations are not yet supported");
+      // TODO: this goes away once residual functions are partially evaluated.
```

Comment:
When functions are partially evaluated, it has to happen after an outer fixed point determines the global state that represents all possible states in which a function can get called.